presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,549 @@ Tuning Presto
+ * **Description:** See :ref:`optimizer.optimize-hash-generation <tuning-pref-optimizer>`.
+
+
+``orc_max_buffer_size``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``hive.orc.max-buffer-size`` (``8 MB``)
+ * **Description:** See :ref:`hive.orc.max-buffer-size <tuning-pref-hive>`.
+
+
+``orc_max_merge_distance``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``hive.orc.max-merge-distance`` (``1 MB``)
+ * **Description:** See :ref:`hive.orc.max-merge-distance <tuning-pref-hive>`.
+
+
+``orc_stream_buffer_size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``hive.orc.max-buffer-size`` (``8 MB``)
+ * **Description:** See :ref:`hive.orc.max-buffer-size <tuning-pref-hive>`.
+
+
+``plan_with_table_node_partitioning``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``true``
+ * **Description:** **Experimental.** Adapt plan to use backend partitioning. By setting this property you allow to use partitioning provided by table layout itself while collecting required data. This may allow to utilize optimization of table layout provided by specific connector. In particular, when this is set presto will try to partition data for workers in a way that each workers gets a chunk of data that comes from one backend partition. It can be particularly useful due to the I/O distribution optimization in table partitioning. Note that this property may only be utilized if given projection uses all columns used for table partitioning inside connector.
+
+
+``prefer_streaming_operators``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Prefer source table layouts that produce streaming operators. Setting this property will allow workers not to wait for chunks of data to start processing them while scanning tables. This may cause faster processing  with lower latency and downtime but some operators may do things more efficiently when working with chunks of data.
+
+
+``push_table_write_through_union``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``optimizer.push-table-write-through-union`` (``true``)
+ * **Description:** See :ref:`optimizer.push-table-writethrough-union <tuning-pref-optimizer>`.
+
+
+``query_max_memory``
+^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``query.max-memory`` (``20 GB``)
+ * **Description:** This property can be use to be nice to the cluster if a particular query is not as important as the usual cluster routines. Setting this value to less than the server property ``query.max-memory`` will cause presto to drop the query in the session if it will require more then ``query_max_memory`` memory. Setting this value to higher than ``query.max-memory`` will not have any effect.
+
+
+
+``query_max_run_time``
```

Comment:
IIRC there is now a `query_max_cpu_time` which is useful for prevent bad queries from a user from using too many resources.